src/scipp/_binding.py

```diff
@@ -30,6 +30,19 @@ def bind_get():
         setattr(cls, 'get', method)
 
 
+def _int_dunder(self) -> int:
+    return int(self.value)
```

Comment:
Shouldn't we also assert that self is dimensionless and has no variance?

Reply:
Good point! We should definitely check the variance. But I am not sure about the unit. `var.value` doesn't check either.

Reply:
Why would `var.value` check the unit? It returns the content of the `value` field? In contrast, what is added here converts a variable to a float, but `1.2 meter` is not a float.